sentry-core/src/metrics/normalization/normalized_tags.rs

```diff
@@ -62,13 +64,13 @@ impl<'a> NormalizedTags<'a> {
 
 impl std::fmt::Display for NormalizedTags<'_> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let res = self
-            .tags
-            .iter()
-            .map(|(k, v)| format!("{}:{}", k, v))
-            .sorted()
-            .join(",");
-        write!(f, "{res}")
+        for (i, (k, v)) in self.tags.iter().enumerate() {
+            if i > 0 {
+                f.write_char(',')?;
+            }
+            write!(f, "{k}:{v}")?;
+        }
+        Ok(())
```

Comment:
The problem with using `format!`, `collect`, and `join` is that they all allocate. Your version would:
1. allocate a `String` for each tag (the `format!` call)
2. collect all those strings into a newly allocated `Vec` (the `collect` call)
3. join the vector into a newly allocated `String` (the `join` call)
4. write that `String` into the formatter

None of these allocations are necessary. We just want to write some already existing strings into a formatter, separated by commas and colons.

Reply:
The primary change here was also switching from a `HashMap` to a `BTreeMap`, to avoid having to pull in the `itertools` dependency to sort the map output, as that happens implicitly now.